tensorflow/core/kernels/scatter_nd_op.cc

```diff
@@ -359,13 +331,20 @@ class ScatterNdUpdateOp : public OpKernel {
                               .HostMemory("shape"),                   \
                           ScatterNdOp<dev##Device, type, index_type>)
 
+#ifdef TENSORFLOW_USE_SYCL
+#define USE_HOST_MEMORY_IF_SYCL(name) .HostMemory(name)
+#else
+#define USE_HOST_MEMORY_IF_SYCL(name)
+#endif  // TENSORFLOW_USE_SYCL
+
```

Comment:
The macro USE_HOST_MEMORY_IF_SYCL should move to another file if we want to use this trick elsewhere.
I think the best would be a new file in tensorflow/core/common_runtime/sycl/ to avoid unnecessary includes.